src/dataEntities/block.ts

```diff
@@ -10,6 +12,18 @@ export interface Logs {
     logs: ethers.providers.Log[];
 }
 
+/**
+ * Returns true the `block` contains a log that matches `filter`, false otherwise.
+ */
+export function hasLogMatchingEventFilter(block: Logs, filter: ethers.EventFilter): boolean {
+    if (!filter.address) throw new ArgumentError("The filter must provide an address");
+    if (!filter.topics) throw new ArgumentError("The filter must provide the topics");
+
+    return block.logs.some(
+        log => log.address === filter.address && filter.topics!.every((topic, idx) => log.topics[idx] === topic)
+    );
+}
```

Comment:
Actually, all the string comparisons here should normalize the strings, e.g. call `toLowercase`.

Reply:
Fixed in e329e67.